NKL/Oper.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Govereau
+-/
+
+import NKL.NKI
+
+namespace NKL
+
+-- Some experimental definitions of semantics for expresssions
+-- For now, I am abstracting over the types of tensors and
+-- tensor operators.
+
+-- Evaluating expressions results in values, which
+-- may contain tensors and tensor operators
+inductive Value (tensor op : Type) where
+  | const (c : Const)
+  | tensor (t : tensor) (ix : List (Value tensor op))
+  | slice (l u step : Value tensor op)
```

Comment:
Chatted offline. `step` could be an int here, but in NumPy is also allowed to be a singleton "scalar" array like `np.array(1)`. Not sure which type is better here.

Reply:
I guess, in the current syntax, this might be written like so: `u=np.array(1); t[1:2:u]`. It is weird to have just `u` and not `u[0]`. Will need to consider how to handle this, well.